contracts/vault/src/execute.rs

```diff
@@ -100,13 +99,9 @@ fn execute_request_withdrawal(
         amount,
     };
 
-    state::WITHDRAWAL_QUEUE.update(
-        deps.storage,
-        |mut queue| -> Result<Vec<WithdrawalRequest>> {
-            queue.push(withdrawal_request);
-            Ok(queue)
-        },
-    )?;
+    let queue_id = get_and_increment_queue_id(deps.storage)?;
+
+    state::WITHDRAWAL_QUEUE.save(deps.storage, queue_id, &withdrawal_request)?;
```

Comment:
One thing that I thought of just now: when `withdrawal_request.amount` is zero, the code should return an error because otherwise this gets added to the queue and in `execute_redistribute_funds` when attempting to send to the user, the TX will result in an error because the VM won't allow sending 0 amounts.